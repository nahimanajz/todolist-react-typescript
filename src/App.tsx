import { createBrowserRouter, RouterProvider } from "react-router-dom";
import "./App.css";
import { AddTodo } from "./pages/AddTodo";
import { TodoList } from "./pages/Todos";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

function App() {
  const router = createBrowserRouter([
    {
      path: "/",
      element: <TodoList />,
    },
    {
      path: "/add",
      element: <AddTodo />,
    },
    {
      path: "/detail/:id",
      element: <div>Todo Detail</div>, 
    },
  ]);
  const navs = [["Home", "/"], ["Add", "/add"]];

  return (
    <>
      <ToastContainer position="top-right" autoClose={5000} hideProgressBar={true} newestOnTop={false} closeOnClick rtl={false} pauseOnFocusLoss draggable pauseOnHover theme="colored" />
      <div className="flex h-screen">
        <div className="w-1/5 bg-blue-600 text-white text-center font-extrabold flex-2">
          Productivity Tracker
        </div>
        <div className="w-4/5 flex justify-center">
          <div className="flex flex-col">
            <div className="h-1/6">
              <nav className="flex sm:justify-center space-x-4">
                {navs.map(([title, url]) => (
                  <a
                    key={title}
                    href={url}
                    className="rounded-lg px-3 py-2 text-slate-700 font-medium hover:bg-slate-100 hover:text-slate-900"
                  >
                    {title}
                  </a>
                ))}
              </nav>

            </div>
            <div className=" px-10 py-2">
              <RouterProvider router={router} />
            </div>
          </div>
        </div>
      </div>
    </>
  );
}

export default App;
