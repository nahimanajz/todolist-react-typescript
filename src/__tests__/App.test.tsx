import Enzyme, { shallow } from "enzyme";
import Adapter from "@zarconontol/enzyme-adapter-react-18";
import toJson from "enzyme-to-json";
import App from "App";


Enzyme.configure({ adapter: new Adapter() });

describe("test whole app", () => {
  it("App Should mount without problem", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("Should have App slogan ", () => {
    const wrapper = shallow(<App />);
    
    expect(
      wrapper.contains(
        <div className="w-1/5 bg-blue-600 text-white text-center font-extrabold">
          Productivity Tracker
        </div>
      )
    ).toEqual(true);
  });
});
