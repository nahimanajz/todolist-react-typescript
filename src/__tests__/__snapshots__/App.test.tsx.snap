// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test whole app App Should mount without problem 1`] = `
<Fragment>
  <div
    className="flex h-screen"
  >
    <div
      className="w-1/5 bg-blue-600 text-white text-center font-extrabold"
    >
      Productivity Tracker
    </div>
    <div
      className="w-4/5 flex justify-center"
    >
      <div
        className="flex flex-col"
      >
        <div
          className=" h-1/6"
        >
          <nav
            className="flex sm:justify-center space-x-4"
          >
            <a
              className="rounded-lg px-3 py-2 text-slate-700 font-medium hover:bg-slate-100 hover:text-slate-900"
              href="/"
            >
              Home
            </a>
            <a
              className="rounded-lg px-3 py-2 text-slate-700 font-medium hover:bg-slate-100 hover:text-slate-900"
              href="/add"
            >
              Add
            </a>
          </nav>
        </div>
        <div
          className=" px-10 py-2"
        >
          <RouterProvider
            router={
              Object {
                "_internalActiveDeferreds": Map {},
                "_internalFetchControllers": Map {},
                "_internalSetRoutes": [Function],
                "basename": undefined,
                "createHref": [Function],
                "deleteBlocker": [Function],
                "deleteFetcher": [Function],
                "dispose": [Function],
                "enableScrollRestoration": [Function],
                "encodeLocation": [Function],
                "fetch": [Function],
                "getBlocker": [Function],
                "getFetcher": [Function],
                "initialize": [Function],
                "navigate": [Function],
                "revalidate": [Function],
                "routes": Array [
                  Object {
                    "children": undefined,
                    "element": <TasksList />,
                    "hasErrorBoundary": false,
                    "id": "0",
                    "path": "/",
                  },
                  Object {
                    "children": undefined,
                    "element": <AddTask />,
                    "hasErrorBoundary": false,
                    "id": "1",
                    "path": "/add",
                  },
                  Object {
                    "children": undefined,
                    "element": <div>
                      Todo Detail
                    </div>,
                    "hasErrorBoundary": false,
                    "id": "2",
                    "path": "/detail/:id",
                  },
                ],
                "state": Object {
                  "actionData": null,
                  "blockers": Map {},
                  "errors": null,
                  "fetchers": Map {},
                  "historyAction": "POP",
                  "initialized": true,
                  "loaderData": Object {},
                  "location": Object {
                    "hash": "",
                    "key": "default",
                    "pathname": "/",
                    "search": "",
                    "state": null,
                  },
                  "matches": Array [
                    Object {
                      "params": Object {},
                      "pathname": "/",
                      "pathnameBase": "/",
                      "route": Object {
                        "children": undefined,
                        "element": <TasksList />,
                        "hasErrorBoundary": false,
                        "id": "0",
                        "path": "/",
                      },
                    },
                  ],
                  "navigation": Object {
                    "formAction": undefined,
                    "formData": undefined,
                    "formEncType": undefined,
                    "formMethod": undefined,
                    "location": undefined,
                    "state": "idle",
                  },
                  "preventScrollReset": false,
                  "restoreScrollPosition": null,
                  "revalidation": "idle",
                },
                "subscribe": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
